File: C# Examples/14 - 15 Creating Add-ins/MyMacroAddIn - Part2/MyMacroAddIn/Installer1.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Configuration.Install;
using System.Linq;
using System.Reflection;
using System.Runtime.InteropServices;

namespace MyMacroAddIn
{
    [RunInstaller(true)]
    public partial class Installer1 : System.Configuration.Install.Installer
    {
        Assembly asm = Assembly.GetExecutingAssembly();
        public Installer1()
        {
            InitializeComponent();
            BeforeUninstall += new InstallEventHandler(Installer1_BeforeUninstall);
            AfterInstall += new InstallEventHandler(Installer1_AfterInstall);
        }

        public void Installer1_AfterInstall(object sender, InstallEventArgs e)
        {
            RegistrationServices regsrv = new RegistrationServices();
            regsrv.RegisterAssembly(asm, AssemblyRegistrationFlags.SetCodeBase);
        }

        private void Installer1_BeforeUninstall(object sender, InstallEventArgs e)
        {
            RegistrationServices regsrv = new RegistrationServices();
            regsrv.UnregisterAssembly(asm);
        }

    }


}

File: 14 - 15 Creating Add-ins/MyMacroAddIn/MyMacroAddIn/Installer1.vb
Imports System.ComponentModel
Imports System.Configuration.Install
Imports System.Runtime.InteropServices

Public Class Installer1

    Public Sub New()
        MyBase.New()

        'This call is required by the Component Designer.
        InitializeComponent()

        'Add initialization code after the call to InitializeComponent

    End Sub

    Private Sub Installer1_AfterInstall(sender As Object, e As InstallEventArgs) Handles Me.AfterInstall
        Dim regsrv As New RegistrationServices
        regsrv.RegisterAssembly(MyBase.GetType().Assembly, AssemblyRegistrationFlags.SetCodeBase)

    End Sub

    Private Sub Installer1_BeforeUninstall(sender As Object, e As InstallEventArgs) Handles Me.BeforeUninstall
        Dim regsrv As New RegistrationServices
        regsrv.UnregisterAssembly(MyBase.GetType().Assembly)
    End Sub
End Class
